source/magpie/index.d

```diff
@@ -378,12 +386,14 @@ public:
                     try
                     {
                         int ele = to!int(next[i]);
-                        indexing[axis].codes[i] ~= ele;
+                        indexing[axis].codes[i] = indexing[axis].codes[i][0 .. insertPos] ~
+                            ele ~ indexing[axis].codes[i][insertPos .. $];
                     }
                     catch(ConvException e)
                     {
                         indexing[axis].index[i] = to!(string[])(indexing[axis].codes[i]);
-                        indexing[axis].index[i] ~= next[i];
+                        indexing[axis].index[i] = indexing[axis].index[i][0 .. insertPos] ~
+                            next[i] ~ indexing[axis].index[i][insertPos .. $];
                         indexing[axis].codes[i] = [];
                         foreach(j; 0 .. indexing[axis].index[i].length)
                             indexing[axis].codes[i] ~= cast(int)j;
```

Comment:
This should be preallocating with `reserve`

Reply:
This was addressed in a later commit.